includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Connection with WooCommerce's features.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use \WP_Error;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Connection with WooCommerce's "My Account" page.
+ */
+class WooCommerce_My_Account {
+	const BILLING_ENDPOINT             = 'billing';
+	const STRIPE_CUSTOMER_ID_USER_META = '_newspack_stripe_customer_id';
+
+	/**
+	 * Cached Stripe customer ID of the current user.
+	 *
+	 * @var string
+	 */
+	private static $stripe_customer_id = null;
+
+	/**
+	 * Initialize.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public static function init() {
+		add_filter( 'woocommerce_account_menu_items', [ __CLASS__, 'my_account_menu_items' ], 1000 );
+		add_action( 'woocommerce_account_' . self::BILLING_ENDPOINT . '_endpoint', [ __CLASS__, 'render_billing_template' ] );
+		add_action( 'init', [ __CLASS__, 'add_rewrite_endpoints' ] );
+		add_action( 'template_redirect', [ __CLASS__, 'redirect_to_account_details' ] );
+		add_filter( 'woocommerce_save_account_details_required_fields', [ __CLASS__, 'remove_required_fields' ] );
+	}
+
+	/**
+	 * Filter "My Account" items, if Stripe is the donations platform.
+	 *
+	 * @param array $items Items.
+	 */
+	public static function my_account_menu_items( $items ) {
+		if ( ! Donations::is_platform_stripe() ) {
+			return $items;
+		}
+		$disabled_wc_menu_items = [ 'dashboard', 'downloads', 'members-area', 'subscriptions', 'edit-address', 'orders', 'payment-methods' ];
```

Comment:
One thing to mention...not many publishers are using it because it's still experimental, but the Newspack Listings plugin's "self-serve" features rely on the Orders and Subscriptions menus for managing purchased listings. We might want to discuss how both feature sets can properly coexist, but in the meantime we can simply leave Orders and Subscriptions menu items active by doing something like:

```php
		if ( class_exists( '\Newspack_Listings\Products' ) && \Newspack_Listings\Products::is_active() ) {
			$disabled_wc_menu_items = array_values(
				array_filter(
					$disabled_wc_menu_items,
					function( $key ) {
						return 'orders' !== $key && 'subscriptions' !== $key;
					}
				)
			);
		}
```

or by applying a filter to the `$disabled_wc_menu_items` variable so the Listings plugin and others can allow their required menu items to remain.